cdap-ranger/cdap-ranger-binding/src/main/java/co/cask/cdap/security/authorization/ranger/binding/RangerAuthorizer.java

```diff
@@ -0,0 +1,301 @@
+      rangerPlugin = new RangerBasePlugin("cdap", "cdap");
+    }
+    rangerPlugin.init();
+    RangerDefaultAuditHandler auditHandler = new RangerDefaultAuditHandler();
+    rangerPlugin.setResultProcessor(auditHandler);
+  }
+
+  @Override
+  public void enforce(EntityId entity, Principal principal, Action action) throws Exception {
+    LOG.debug("Enforce called on entity {}, principal {}, action {}", entity, principal, action);
+    if (rangerPlugin == null) {
+      throw new RuntimeException("CDAP Ranger Authorizer is not initialized.");
+    }
+
+    if (principal.getType() != Principal.PrincipalType.USER) {
+      throw new IllegalArgumentException(String.format("The principal type for current enforcement request is '%s'. " +
+                                                         "Authorization enforcement is only supported for '%s'.",
+                                                       principal.getType(), Principal.PrincipalType.USER));
+    }
+    String requestingUser = principal.getName();
+    String ip = InetAddress.getLocalHost().getHostName();
+    Set<String> userGroups = MiscUtil.getGroupsForRequestUser(requestingUser);
+
+    LOG.debug("Requesting user {}, ip {}, requesting user groups {}", requestingUser, ip, userGroups);
+
+    Date eventTime = new Date();
+    String accessType = toRangerAccessType(action);
+    RangerAccessRequestImpl rangerRequest = new RangerAccessRequestImpl();
+    rangerRequest.setUser(requestingUser);
+    rangerRequest.setUserGroups(userGroups);
+    rangerRequest.setClientIPAddress(ip);
+    rangerRequest.setAccessTime(eventTime);
+
+    RangerAccessResourceImpl rangerResource = new RangerAccessResourceImpl();
+    rangerRequest.setResource(rangerResource);
+    rangerRequest.setAccessType(accessType);
+    rangerRequest.setAction(accessType);
+    rangerRequest.setRequestData(entity.toString());
+
+    setAccessResource(entity, rangerResource);
+
+    boolean isAuthorized = false;
+
+    try {
+      RangerAccessResult result = rangerPlugin.isAccessAllowed(rangerRequest);
+      if (result == null) {
+        LOG.warn("Unauthorized: Ranger Plugin returned null for this authorization enforcement.");
+        isAuthorized = false;
+      } else {
+        isAuthorized = result.getIsAllowed();
+      }
+    } catch (Throwable t) {
+      LOG.warn("Error while calling isAccessAllowed(). request {}", rangerRequest, t);
+      throw t;
+    } finally {
+      LOG.trace("Ranger Request {}, authorization {}.", rangerRequest, (isAuthorized ? "successful" : "failed"));
+    }
+
+    if (!isAuthorized) {
+      throw new UnauthorizedException(principal, action, entity);
+    }
+  }
+
+  @Override
+  public void enforce(EntityId entityId, Principal principal, Set<Action> set) throws Exception {
+    LOG.debug("Enforce called on entity {}, principal {}, actions {}", entityId, principal, set);
+    //TODO: Investigate if its possible to make the enforce call with set of actions rather than one by one
```

Comment:
That will be done in another PR. We don't know the exact behavior of it. It needs through testing.